### PR TITLE
User

### PR DESCRIPTION
**Pull request by HarshnaNHaswani**
- Profile page
- Change Password feature
profile
profile2
profile3

### CODE REVIEW

File: src/pages/User/ChangePassword.jsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  const [password, setPassword] = useState(initialPasswordState);
+
+  const updateAndClean = () => {
+    setPasswordDialog(false);
+    userLoading("Processing password change request");
+    changePassword(password.new);
+    setPassword(initialPasswordState);
+  };
+
+  const validatePassword = () => {
```
Comment (shlok30) [inline src/pages/User/ChangePassword.jsx:23]: Very well written function to handle all the error cases

File: src/pages/User/ChangePassword.jsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
+import React, { useState } from "react";
+import { useUser } from "context/user-context.js";
+import userStyles from "./userStyles.module.css";
+
+export const ChangePassword = ({ setPasswordDialog }) => {
+  const { userDataState, changePassword, userLoading } = useUser();
+  const { user, error, loading } = userDataState;
+  const initialPasswordState = {
```
Comment (shlok30) [inline src/pages/User/ChangePassword.jsx:8]: Nice

File: src/reducers/userReducer.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    case "SET_USER": {
+      return {
+        ...state,
+        user: { ...action.payload.userData },
+        error: "",
+        loading: '',
+      };
+    }
+
+    case "CHANGE_PASSWORD": {
```
Comment (shlok30) [inline src/reducers/userReducer.js:12]: Easy to understand cases

**Review by shlok30 — APPROVED** [review]
LGTM

**Comment by HarshnaNHaswani** [comment]
Thank you @shlok30!

**Pull request merged**